File: frameworks/Rust/roa/src/endpoints.rs
use crate::utils::{Message, JSON_LEN, PLAINTEXT_LEN};
use roa::http::header::CONTENT_LENGTH;
use roa::preload::*;
use roa::{Context, Result};

static HELLO_WORLD: &str = "Hello, World!";

#[inline]
pub async fn json(ctx: &mut Context) -> Result {
    ctx.resp.headers.insert(CONTENT_LENGTH, JSON_LEN.clone());
    ctx.write_json(&Message {
        message: HELLO_WORLD,
    })
}

#[inline]
pub async fn plaintext(ctx: &mut Context) -> Result {
    ctx.resp
        .headers
        .insert(CONTENT_LENGTH, PLAINTEXT_LEN.clone());
    ctx.write(HELLO_WORLD);
    Ok(())
}

File: frameworks/CSharp/celerio/src/Endpoints.cs
namespace src;

using Celerio;
using static Celerio.Result;

public static class Endpoints
{
    [Get("/plaintext")]
    public static Result Plaintext()
    {
        return Ok().Text("Hello, World!");
    }

    [Serializable]
    public class SampleResponse
    {
        public string message { get; set; } = "Hello, World!";
    }
    
    [Get("/json")]
    public static Result Json()
    {
        return Ok().Json(new SampleResponse());
    }
}